packages/flutter_tools/fuchsia_attach.dart

```diff
@@ -0,0 +1,128 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:args/args.dart';
+
+import 'package:flutter_tools/runner.dart' as runner;
+import 'package:flutter_tools/src/artifacts.dart';
+import 'package:flutter_tools/src/base/common.dart';
+import 'package:flutter_tools/src/base/context.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/cache.dart';
+import 'package:flutter_tools/src/commands/attach.dart';
+import 'package:flutter_tools/src/commands/doctor.dart';
+import 'package:flutter_tools/src/fuchsia/fuchsia_sdk.dart';
+import 'package:flutter_tools/src/run_hot.dart';
+import 'package:flutter_tools/src/runner/flutter_command.dart';
+
+final ArgParser parser = ArgParser()
+  ..addOption('build-dir', help: 'The fuchsia build directory')
+  ..addOption('dart-sdk', help: 'The prebuilt dart SDK')
+  ..addOption('target', help: 'The GN target to attach to')
+  ..addFlag('verbose', negatable: true);
+
+// Track the original working directory so that the tool can find the
+// flutter repo in third_party.
+String originalWorkingDirectory;
+
+Future<void> main(List<String> args) async {
+  final ArgResults argResults = parser.parse(args);
+  final bool verbose = argResults['verbose'];
+  final String target = argResults['target'];
+  final List<String> targetParts = _extractPathAndName(target);
+  final String path = targetParts[0];
+  final String name = targetParts[1];
+  final File dartSdk = fs.file(argResults['dart-sdk']);
+  final String buildDirectory = argResults['build-dir'];
+  final File frontendServer = fs.file('$buildDirectory/host_x64/gen/third_party/flutter/frontend_server/frontend_server_tool.snapshot');
```

Comment:
This should probably be controlled by a flag

Reply:
scratch that... then all of these would be as well.  I guess this is fine to be tightly coupled to Fuchsia tree structure.